opensrp-reveal/src/test/java/org/smartregister/reveal/util/InteractorUtilsTest.java

```diff
@@ -0,0 +1,179 @@
+package org.smartregister.reveal.util;
+
+import net.sqlcipher.Cursor;
+import net.sqlcipher.MatrixCursor;
+import net.sqlcipher.database.SQLiteDatabase;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.powermock.reflect.Whitebox;
+import org.smartregister.domain.db.Event;
+import org.smartregister.domain.db.EventClient;
+import org.smartregister.repository.BaseRepository;
+import org.smartregister.repository.EventClientRepository;
+import org.smartregister.repository.TaskRepository;
+import org.smartregister.reveal.BaseUnitTest;
+import org.smartregister.reveal.model.BaseTaskDetails;
+import org.smartregister.reveal.sync.RevealClientProcessor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.smartregister.reveal.util.Constants.BEHAVIOUR_CHANGE_COMMUNICATION;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.BASE_ENTITY_ID;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.CASE_CONFIRMATION_FIELD;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.FORM_SUBMISSION_ID;
+import static org.smartregister.reveal.util.Constants.Intervention.BCC;
+import static org.smartregister.reveal.util.Constants.Intervention.BLOOD_SCREENING;
+import static org.smartregister.reveal.util.Constants.Intervention.CASE_CONFIRMATION;
+
+/**
+ * Created by Richard Kareko on 2/18/20.
+ */
+
+public class InteractorUtilsTest extends BaseUnitTest {
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Mock
+    private TaskRepository taskRepository;
+
+    @Mock
+    private EventClientRepository eventClientRepository;
+
+    @Mock
+    private RevealClientProcessor clientProcessor;
+
+    @Mock
+    private RevealJsonFormUtils jsonFormUtils;
+
+    @Mock
+    SQLiteDatabase database;
+
+    @Captor
+    ArgumentCaptor<JSONArray> jsonArrayArgumentCaptor;
+
+    @Captor
+    ArgumentCaptor<Long> longArgumentCaptor;
+
+    @Captor
+    ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Captor
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4997357)